### ISSUE TITLE
Training with unsloth 

### ISSUE THREAD

**Issue by ZetangForward** [FlagOpen/FlagEmbedding#919]
Currently, Unsloth can only support single GPU training, how can you implement it with 8-GPU training? Thx

**Comment by namespace-Pt** [comment]
Hi, it's weird that unsloth was able to use DDP two months ago :anguished:. Maybe you should wait newer version of unsloth or use some LLM framework like Megatron for efficient training. I'll update the README...

We choosed to use unsloth because it can efficiently tune the Llama-3 model given its huge vocab size while the native implementation of huggingface cannot. An alternative is to break the loss computation of long sequences into parts and  manually implement the backward function to avoid instantiating the entire `seq_len x vocab_size` table in GPU memory.

**Comment by namespace-Pt** [comment]
Try the following?
```bash
CUDA_VISIBLE_DEVICES=0 torchrun --nproc_per_node 1 -m main.train \
--data_root /data/long-llm \
--output_dir data/outputs/$output_name \
--model_name_or_path meta-llama/Meta-Llama-3-8B-Instruct \
--train_data long-llm:gpt/one_detail_book.train.64K.json long-llm:gpt/one_detail_paper.train.64K.json long-llm:gpt/multi_detail_book.train.json long-llm:gpt/multi_detail_paper_short.train.json long-llm:gpt/multi_detail_paper_long.train.json long-llm:gpt/bio_book.train.json long-llm:longalpaca/train.json long-llm:redpajama/train.json[5000] \
--max_length 81920 \
--group_by_length \
--rope_theta 200e6 \
--attn_impl flash_attention_2 \
--gradient_checkpointing \
--use_reentrant True \
--learning_rate 5e-5 \
--num_train_epochs 1 \
--save_only_model \
--save_strategy epoch \
--logging_steps 5 \
--bf16 \
--lora_tune \
--lora_extra_params embed_tokens \
--load_in_4_bit \
--chat_template llama-3
```

**Comment by muratsilahtaroglu** [comment]
I solved this error because it happens when I install or want to train another model cuda at the same time. It worked when I closed other applications.
